instrumentation/src/main/java/io/opentelemetry/android/internal/features/persistence/SimpleTemporaryFileProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.features.persistence;
+
+import io.opentelemetry.contrib.disk.buffering.internal.files.TemporaryFileProvider;
+import java.io.File;
+
+/**
+ * This class is internal and not for public use. Its APIs are unstable and can change at any time.
+ */
+public final class SimpleTemporaryFileProvider implements TemporaryFileProvider {
+    private final File tempDir;
+
+    public SimpleTemporaryFileProvider(File tempDir) {
+        this.tempDir = tempDir;
+    }
+
+    @Override
+    public File createTemporaryFile(String prefix) {
```

Comment:
The name suggests that maybe the file is actually _created_ on disk. Worth either a clarifying comment or rename to `getTemporaryFile()`? I guess it is an interface method...hmmmm... 🤔

Reply:
Yeah fair enough, I had overlooked that the interface lived upstream. It's fine for now, but still a comment could be nice.